File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer = []
    days = []
    for i in range(len(progresses)):
        days.append(math.ceil((100 - progresses[i])/speeds[i]))
    
    cnt = 1
    pre = days[0]
    for i in range(1, len(days)):
        if days[i] <= pre:
            cnt += 1
        else:
            answer.append(cnt)
            cnt = 1
            pre = days[i]
    answer.append(cnt)

        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> arr = new ArrayList<>();
        int N = progresses.length;
        
        int day = 0;
        int cnt = 0;
        for (int i = 0; i < N; i++) {
            int x = 100 - progresses[i];
            int y = x / speeds[i];
            if (x % speeds[i] != 0) y++;
            if (day < y) {
                arr.add(cnt);
                cnt = 1;
                day =y;
            } else {
                cnt++;
            }
        }
        arr.add(cnt);
        int[] answer = new int[arr.size() - 1];
        
        for (int i = 1; i < arr.size(); i++) {
            answer[i - 1] = arr.get(i);
        }
        
        return answer;
    }
}